src/main/java/com/fouristhenumber/utilitiesinexcess/common/items/ItemInversionSigilActive.java

```diff
@@ -2,29 +2,500 @@
+                player.addChatMessage(
+                    new ChatComponentTranslation(
+                        StatCollector.translateToLocalFormatted(
+                            "chat.pseudo_inversion_ritual.onequarter",
+                            source.siegeMobsKilled)));
+            }
+            return;
+        }
+
+        if (!(event.entityLiving instanceof EntityIronGolem)) {
+            return;
+        }
+
+        if (!player.inventory.hasItem(this)) return;
+
+        int radius = BEACON_SEARCH_RADIUS;
+        int mobX = (int) Math.floor(event.entityLiving.posX);
+        int mobY = (int) Math.floor(event.entityLiving.posY);
+        int mobZ = (int) Math.floor(event.entityLiving.posZ);
+
+        int beaconX = 0, beaconY = 0, beaconZ = 0;
+        boolean found = false;
+
+        for (int dx = -radius; dx <= radius && !found; dx++) {
+            for (int dy = -2; dy <= 2 && !found; dy++) {
+                for (int dz = -radius; dz <= radius && !found; dz++) {
+                    int bx = mobX + dx;
+                    int by = mobY + dy;
+                    int bz = mobZ + dz;
+                    if (world.getBlock(bx, by, bz) == Blocks.beacon) {
+                        beaconX = bx;
+                        beaconY = by;
+                        beaconZ = bz;
+                        found = true;
+                    }
+                }
+            }
+        }
+
+        if (!found) return;
+
+        boolean dimensionOk = (world.provider.dimensionId == 1);
+        boolean chestNorthContentsOk = false;
+        boolean chestEastContentsOk = false;
+        boolean chestSouthContentsOk = false;
+        boolean chestWestContentsOk = false;
+        boolean spiralOk = checkSpiral(world, beaconX, beaconY, beaconZ);
+
+        if (world.getTileEntity(beaconX, beaconY, beaconZ - 5) instanceof TileEntityChest chest) {
+            chestNorthContentsOk = checkChest(chest, CHEST_NORTH_CONTENTS, InversionConfig.northChestRequiredItems);
+        }
+        if (world.getTileEntity(beaconX + 5, beaconY, beaconZ) instanceof TileEntityChest chest) {
+            chestEastContentsOk = checkChest(chest, CHEST_EAST_CONTENTS, InversionConfig.eastChestRequiredItems);
+        }
+        if (world.getTileEntity(beaconX, beaconY, beaconZ + 5) instanceof TileEntityChest chest) {
+            chestSouthContentsOk = checkChest(chest, CHEST_SOUTH_CONTENTS, InversionConfig.southChestRequiredItems);
+        }
+        if (world.getTileEntity(beaconX - 5, beaconY, beaconZ) instanceof TileEntityChest chest) {
+            chestWestContentsOk = checkChest(chest, CHEST_WEST_CONTENTS, InversionConfig.westChestRequiredItems);
+        }
+
+        if (!(dimensionOk && spiralOk
+            && chestNorthContentsOk
+            && chestEastContentsOk
+            && chestSouthContentsOk
+            && chestWestContentsOk)) {
+            return;
+        }
```

Comment:
You could turn all of these into early returns.
```suggestion
        if (world.provider.dimensionId != 1) return;
        if (!checkSpiral(world, beaconX, beaconY, beaconZ)) return;

        if (!(world.getTileEntity(beaconX, beaconY, beaconZ - 5) instanceof TileEntityChest northChest)
            || !checkChest(northChest, CHEST_NORTH_CONTENTS, InversionConfig.northChestRequiredItems)) {
            return;
        }
        if (!(world.getTileEntity(beaconX + 5, beaconY, beaconZ) instanceof TileEntityChest eastChest)
            || !checkChest(eastChest, CHEST_EAST_CONTENTS, InversionConfig.eastChestRequiredItems)) {
            return;
        }
        if (!(world.getTileEntity(beaconX, beaconY, beaconZ + 5) instanceof TileEntityChest southChest)
            || !checkChest(southChest, CHEST_SOUTH_CONTENTS, InversionConfig.southChestRequiredItems)) {
            return;
        }
        if (!(world.getTileEntity(beaconX - 5, beaconY, beaconZ) instanceof TileEntityChest westChest)
            || !checkChest(westChest, CHEST_WEST_CONTENTS, InversionConfig.westChestRequiredItems)) {
            return;
        }
```